### ISSUE TITLE
[DOCS ISSUE] Installing IPFS should be the main thing about https://docs-beta.ipfs.io/install/ 

### ISSUE THREAD

**Issue by hsanjuan** [ipfs/docs#465]
**URL of the page in question:**
https://docs-beta.ipfs.io/install/

**What's wrong with this page?**

Clicking on "install IPFS" takes me to a page where the big Headers point to installing Companion, Desktop, Cluster and at the very bottom, actually, IPFS.

(there is a link at the top that I only saw when I was actually forced to read the thing in detail).

**What would you like to have changed/amended/fixed in order to make things right?**

Have an entry page with links (or a top-level index) that says  "What do you want to do?" (h1) and lists:

* Install the IPFS daemon (CLI application)
* Install IPFS Desktop (Graphical interface)
* Install the browser plugin, IPFS Companion (works best with IPFS Desktop or Daemon)
* Develop applications using IPFS (Link to developer docs for Go/JS)
* Host and share files at scale (Link to Cluster)

Upon clicking you can then explain the subtleties of that choice in their section/page.

I would remove the `Dip your toes in decentralized waters` as the top-level main thing. This suggests companion is the right tool but I'm not sure if IPFS companion offers a good experience without installing go-ipfs in the first place. I am guessing it will rely on embedded js-ipfs which does not have dht and can only find content if the preloaders are working well, so it might be suboptimal to make it the top-level thing you need to install if you want to install IPFS (IMHO).

**Comment by johnnymatthews** [comment]
There's a bunch of different ways to _Install IPFS_, either through the desktop app, the terminal, or using something like the browser companion.  Something like the following might be more suitable:

```
Install IPFS			<---- Section.
	Desktop Application	<---- Page.
		Windows		<---- ## H2 header in page.
		MacOS
		Linux
	Terminal/Command-line
		Windows
		MacOS
		Linux
	Browser Companion
		Chrome
		Firefox
```

The Desktop Application section is already being produced in [`ipfs/ipfs-docs-v2:website-on-ipfs`](https://github.com/ipfs/ipfs-docs-v2/blob/website-on-ipfs/docs/how-to/websites-on-ipfs/host-single-page-site.md#install-ipfs-desktop).

With these pages created, the Install Homepage can be as you suggested @hsanjuan.

**Comment by hsanjuan** [comment]
> There's a bunch of different ways to _Install IPFS_, either through the desktop app, the terminal, or using something like the browser companion. Something like the following might be more suitable:
> 
> ```
> Install IPFS			<---- Section.
> 	Desktop Application	<---- Page.
> 		Windows		<---- ## H2 header in page.
> 		MacOS
> 		Linux
> 	Terminal/Command-line
> 		Windows
> 		MacOS
> 		Linux
> 	Browser Companion
> 		Chrome
> 		Firefox
> ```
> 
> The Desktop Application section is already being produced in [`ipfs/ipfs-docs-v2:website-on-ipfs`](https://github.com/ipfs/ipfs-docs-v2/blob/website-on-ipfs/docs/how-to/websites-on-ipfs/host-single-page-site.md#install-ipfs-desktop).
> 
> With these pages created, the Install Homepage can be as you suggested @hsanjuan.

Yeah, this sounds very good. Also, we could auto-detect the user's OS and show the corresponding instructions. 

An example: https://decred.org/ : shows the canonical download (gui app for my OS) that most users will want to use (I guess this would be IPFS Desktop). But gives me the alternative to go to: https://decred.org/wallets/ where I have other alternatives (CLI, 3rd party, mobile stuff, extensions). Those that deserve it can have multiple OS instructions. Power/CLI users may not need super detailed instructions at all. Non power users are never going to choose the CLI option, so those that do know how to download,unpack, execute the things without much ado.

(we don't have to do this, it may not even be good for our ecosystem, it's just ideas).

**Comment by johnnymatthews** [comment]
> auto-detect the user's OS

I'm assuming this is doable @cwaring? It's not super urgent, but it could definitely be a cool feature to have in the future.

**Comment by jessicaschilling** [comment]
Not directly related to this issue, but eventually I'd like https://dist.ipfs.io/ to be closer to all of the things mentioned above in this thread. I've got this as one of the potential future efforts in our ecosystem audit recommendations: https://ipfs.canny.io/admin/board/ipfs-information-ecosystem-audit-recommendations/p/add-branding-and-context-to-distipfsio

**Comment by cwaring** [comment]
> auto-detect the user's OS

Yes we could do this with a custom vue component, would be a nice add-on 👌

**Comment by hsanjuan** [comment]
> I'd like https://dist.ipfs.io/ to be closer to all of the things mentioned above in this thread

Maybe avoid sending people to dist.ipfs.io altogether and simply link the actual downloads directly. If we have nice docs and ipfs.io page people should probably not have to navigate the dists page at all (and thus docs team does not have to worry about yet another site).

**Comment by johnnymatthews** [comment]
> Maybe avoid sending people to dist.ipfs.io altogether 

For now, yeah this is a good idea.

> simply link the actual downloads directly

I'd much rather link to where they can download package/exe/etc from GitHub, with detailed instructions on what to look for and how they can get to the **Releases** tab themselves. This way we can avoid linking to old packages.

**Comment by jessicaschilling** [comment]
Agreed that it's easier/more effective to put the (automatically generated/automatically updated) list of downloadables either on the docs site or the main ipfs.io, but I'm torn on which one. There's no clear industry pattern for one over the other ... and this actually feeds into a larger issue of whether the main IPFS "sales site" (which is only a few pages) really shouldn't just be a more visually interesting set of pages on the docs site itself.

Right now it's looking like "rework the IPFS site" is going to be one of the top recommendations coming out of some of our info ecosystem audit work, so watch this space, but would love to keep discussing here in the meantime.

**Comment by johnnymatthews** [comment]
> rework the IPFS site

💯 👍 

For now I'd rather put links to where the reader can download stuff from, and not hard link from the docs. But yeah, it would be nice to have common downloads (desktop app, daemon, browser companion) on ipfs.io.

**Comment by jessicaschilling** [comment]
Adding notes based on recent conversation with @johnnymatthews:

Our Q1 ecosystem audit included a highly-ranked recommendation to create a Q&A/flowchart-based "what flavor IPFS" guide. @hsanjuan's original concerns here can be remedied by building such a guide, so let's repurpose this issue for achieving this.

See https://ipfs.canny.io/admin/board/ipfs-information-ecosystem-audit-recommendations/p/qaflowchart-based-what-flavor-ipfs-guide for more details.

**Comment by DavidBurela** [comment]
I had a similar concern about the quickstart page, where the install is talking about building from source, instead of how to install go-ipfs or ipfs-update. 
#494

**Comment by DavidBurela** [comment]
Here is a good sample of what I expect when I go to an install page for a project
https://helm.sh/docs/intro/install/

even just some bash commands on how to pull down and put into /usr/local/bin, or ipfs-update

**Comment by johnnymatthews** [comment]
This is an important point @DavidBurela, thanks for bringing it up. The entire point of a _Quick start_ to for the reader to get their hands dirty tinkering around with things as soon as possible. 

While the command line is the best option for some readers, a more likely _easy-path_ for everyone would be to install the IPFS Desktop App. It's simple to install, and even simpler to use. Towards the end of the _Quick start_ we can mention how to quickly set up the IPFS daemon for both Go and JS.

**Comment by hsanjuan** [comment]
I just saw that `ipfs-update` is not documented as the simplest way to install, upgrade, downgrade go-ipfs. This needs porting: http://docs.ipfs.io.ipns.localhost:8080/guides/guides/install/#installing-with-ipfs-update ?

**Comment by johnnymatthews** [comment]
You're right @hsanjuan. The [install](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/install/) and [update](https://docs-beta.ipfs.io/recent-releases/go-ipfs-0-5/update-procedure/) instructions for Go-IPFS 0.5 use `dist.ipfs.io`, but the main `/install` docs do not.

Issue #498 is available for anyone who's got time.

**Comment by hsanjuan** [comment]
Ok that's an issue but I was talking about the `ipfs-update` tool.

**Comment by johnnymatthews** [comment]
Sorry I misunderstood. 

Are you talking about `ipfs-update` specific instructions? There are install and usage instructions within the [`ipfs/ipfs-update` repo](https://github.com/ipfs/ipfs-update/blob/v1.6.0/README.md). We should bring them into `ipfs/docs`. Issue #499 addresses this.

**Comment by hsanjuan** [comment]
The idea would be to make ipfs-update the canonical way to install ipfs (and update, and downgrade). 

Downloading directly from dist becomes a secondary way, more hidden.
